src/StateSnapshot.js

```diff
@@ -356,3 +363,26 @@ function isLiteral(node) {
     memberExpression.property.type === 'NumericLiteral'
   )
 }
+
+function isInJSXContainer(node) {
+  return (
+    isInSomething(node, 'JSXExpressionContainer') ||
+    isInSomething(node, 'JSXElement')
+  )
+}
+
+function isInDeclaration(node) {
+  const _parentDeclaration = getParentOfNodeType(node, 'VariableDeclarator')
+
+  if (
+    _parentDeclaration &&
+    _parentDeclaration.init &&
+    _parentDeclaration.init.callee &&
+    _parentDeclaration.init.callee.name &&
+    _parentDeclaration.init.callee.name === 'useSnapshot'
```

Comment:
We can use optional chaining. I think babel transpiles it if needed.

Reply:
Would also need changing the eslint config, so didn't do it, 
the next commit does handle it

Reply:
Nice.